src/core_modules/capture-core/components/FormFields/New/Fields/OrgUnitField/OrgUnitField.component.js

```diff
@@ -51,61 +47,104 @@ type Props = {
 
-    handleFilterChange = (event: SyntheticEvent<HTMLInputElement>) => {
-        this.props.onSearch(event.currentTarget.value);
-    }
-    render() {
-        const {
-            roots,
-            onSelectClick,
-            searchText,
-            ready,
-            treeKey,
-            classes,
-            selected,
-            maxTreeHeight,
-            disabled,
-        } = this.props;
-        const styles = maxTreeHeight ? { maxHeight: maxTreeHeight, overflowY: 'auto' } : null;
-        return (
-            <div
-                className={classes.container}
-            >
-                <div className={classes.debounceFieldContainer}>
-                    <DebounceField
-                        onDebounced={this.handleFilterChange}
-                        value={searchText}
-                        placeholder={i18n.t('Search')}
-                        classes={this.classes}
-                        disabled={disabled}
-                    />
-                </div>
-                {!disabled &&
-                    <div className={classes.orgUnitTreeContainer} style={styles}>
-                        <OrgUnitTree
-                            roots={roots}
-                            onSelectClick={onSelectClick}
-                            ready={ready}
-                            treeKey={treeKey}
-                            selected={selected}
-                        />
-                    </div>
-                }
+    const { loading, data } = useDataQuery(
+        React.useMemo(
+            () => ({
+                orgUnits: {
+                    resource: 'me',
+                    params: {
+                        fields: ['organisationUnits[id,path]'],
+                    },
+
+                },
+            }),
+            [],
+        ),
+    );
+
+
+    const { loading: searchLoading, data: searchData, refetch: refetchOrg } = useDataQuery(
+        React.useMemo(
+            () => ({
+                orgUnits: {
+                    resource: 'organisationUnits',
+                    params: ({ variables: { searchText: currentSearchText } }) => ({
+                        fields: [
+                            'id,displayName,path,publicAccess,access,lastUpdated',
+                            'children[id,displayName,publicAccess,access,path,children::isNotEmpty]',
+                        ].join(','),
+                        paging: true,
+                        query: currentSearchText,
+                        withinUserSearchHierarchy: true,
+                        pageSize: 15,
+                    }),
+
+                },
+            }),
+            [],
+        ),
+        { lazy: true },
+    );
+
+    const ready = searchText?.length ? !searchLoading : !loading;
+
+    React.useEffect(() => {
+        if (searchText?.length) {
+            refetchOrg({ variables: { searchText } });
+        }
+    }, [refetchOrg, searchText]);
+
+    const getRoots = React.useMemo(() => {
```

Comment:
Hey, 

The error `rootNode is undefined` is thrown when I try to search for an org unit (write or delete some letters in the field). This error occurs in both the ScopeSelector and the forms. 
I think it may be related to the logic of the `getRoots` function. Can you have a look?



https://user-images.githubusercontent.com/6817073/176856318-b1c742b8-1f13-4344-8cfb-e20ed49c4c36.mov

Thank you!